Sources/PackageGraph/PackageGraphLoader.swift

```diff
@@ -222,9 +243,21 @@ private func createResolvedPackages(
         let package = packageBuilder.package
 
         // Establish the manifest-declared package dependencies.
-        packageBuilder.dependencies = package.manifest.package.dependencies.compactMap({
+        let dependencies = package.manifest.package.dependencies.compactMap({
             packageMap[PackageReference.computeIdentity(packageURL: $0.url)]
         })
+        
+        // Detect duplicate dependencies
+        for duplicate in dependencies.findDuplicates() {
```

Comment:
Hmm, it is odd that this would work because findDuplicates works using equality and two `Dependency` objects won't be equal if their version number differs. Also, I think we should find duplicates using package identity here.